### PR TITLE
MS15184: Remove deprecated functions in Agent API

### PR DESCRIPTION
**Pull request by thoys**
Fixes https://highfidelity.manuscript.com/f/cases/15184/Remove-deprecated-functions-in-Agent-h-and-ThreadedAssignment-h 

## QA Test plan
1. Be sure you have the full installer for the PR with sandbox installed.
2. Run the sandbox.
3. Go to http://localhost:40100/content/#scripts_group and add 5 instances for the following persistent script https://raw.githubusercontent.com/highfidelity/hifi/master/script-archive/acScripts/BetterClientSimulationBotFromRecording.js
4. Visit hifi://simulator/103.623,2.2868,36.9127/0,0.998816,0,0.0486569  
5. You should see 5 bots pop up , each of them playing a recording. ( Could take a while for them to load in)
6. Remove the persistent script.
7. Open Server Console logs through the traybar icon, right-click it -> View Logs. Go to the Assignment Clients tab in the logs.
8. Load one instance of https://raw.githubusercontent.com/thoys/hifi/e2d7431456025f38a1f2c9941d81e27ff61da830/scripts/developer/tests/agentAPITest.js as a persistent script.
9. The log should contain the following lines:
    ```
    [07/28 03:20:37] [INFO] [hifi.scriptengine] [28300] [agent] [agentAPITest.js] Script Engine starting:agentAPITest.js
    [07/28 03:20:37] [WARNING] [default] [28300] [agent] DependencyManager::get(): No instance available for class Midi
    [07/28 03:20:37] [WARNING] [hifi.scriptengine] [28300] [agent] [agentAPITest.js] Agent.isAvatar = false
    [07/28 03:20:37] [WARNING] [hifi.scriptengine] [28300] [agent] [agentAPITest.js] Agent.isAvatar = true
    [07/28 03:20:37] [WARNING] [hifi.scriptengine] [28300] [agent] [agentAPITest.js] Agent.isListeningToAudioStream = false
    [07/28 03:20:37] [WARNING] [hifi.scriptengine] [28300] [agent] [agentAPITest.js] Agent.isListeningToAudioStream = true
    [07/28 03:20:37] [WARNING] [hifi.scriptengine] [28300] [agent] [agentAPITest.js] Agent.isNoiseGateEnabled = false
    [07/28 03:20:37] [WARNING] [hifi.scriptengine] [28300] [agent] [agentAPITest.js] Agent.isNoiseGateEnabled = true
    [07/28 03:20:37] [WARNING] [hifi.scriptengine] [28300] [agent] [agentAPITest.js] Agent.lastReceivedAudioLoudness = 8.407790785948902e-45
    [07/28 03:20:37] [WARNING] [hifi.scriptengine] [28300] [agent] [agentAPITest.js] Agent.sessionUUID = {4b153a09-80e0-4000-9b51-de3caef76766}
    [07/28 03:20:37] [DEBUG] [hifi.audio] [28300] [agent] Requesting sound at "http://hifi-content.s3.amazonaws.com/howard/sounds/piano1.wav"
    [07/28 03:20:37] [DEBUG] [hifi.networking.resource] [28300] [agent] Starting request for: http://hifi-content.s3.amazonaws.com/howard/sounds/piano1.wav
    ````
    - Numberic/date/uuid values may differ
    - boolean values should match ( true / false )

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11516/HighFidelity-Beta-PR13676-e2d7431.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11516/HighFidelity-Beta-PR13676-e2d7431.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11516/HighFidelity-Beta-Interface-PR13676-e2d7431.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11516/HighFidelity-Beta-PR13676-e2d7431.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11516/HighFidelity-Beta-Interface-PR13676-e2d7431.exe

**Comment by qlipdahlin** [comment]
Currently Testing

**Comment by qlipdahlin** [comment]
@conklin94122 PASS

Loaded sandbox, created 5 instances of bots on DS, verified bots were present, removed bots, opened server console logs, loaded agentAPItest script, verified in logs all items listed with correct boolean values.

Checked on both PC and Mac, both saw all related items in logs. Including screenshot for most of the boolean logs.
capture01

File: assignment-client/src/Agent.cpp [hunk 1]
```diff
@@ -452,7 +453,7 @@ void Agent::executeScript() {
     packetReceiver.registerListener(PacketType::AvatarIdentity, avatarHashMap.data(), "processAvatarIdentityPacket");
 
     // register ourselves to the script engine
-    _scriptEngine->registerGlobalObject("Agent", this);
+    _scriptEngine->registerGlobalObject("Agent", new AgentScriptingInterface(this));
```
Comment (huffman) [inline assignment-client/src/Agent.cpp]: This looks like a leak - will this get cleaned up anywhere?

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11663/HighFidelity-Beta-PR13676-92efc1a.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11663/HighFidelity-Beta-PR13676-92efc1a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11663/HighFidelity-Beta-Interface-PR13676-92efc1a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11663/HighFidelity-Beta-PR13676-92efc1a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11663/HighFidelity-Beta-Interface-PR13676-92efc1a.exe

File: assignment-client/src/Agent.cpp (hunk 1 above)
Comment (thoys) [reply assignment-client/src/Agent.cpp]: Took care of it using the suggested QObject constructor parent parameter method

**Comment by qlipdahlin** [comment]
@conklin94122 PASS

Loaded sandbox, created 5 instances of bots on DS, verified bots were present, removed bots, opened server console logs, loaded agentAPItest script, verified in logs all items listed with correct boolean values. Saw all relevant items in logs.

**Pull request merged**